### PR TITLE
Disable msccl at compile time

### PR DESCRIPTION
**Pull request by BertanDogancay**

### CODE REVIEW

**Review by wenkaidu — DISMISSED** [review]

File: install.sh [hunk 1]
```diff
@@ -208,6 +211,10 @@ if [[ "${collective_trace}" == false ]]; then
     cmake_common_options="${cmake_common_options} -DCOLLTRACE=OFF"
 fi
 
+if [[ "${msccl_enabled}" == false ]]; then
+    cmake_common_options="${cmake_common_options} -DENABLE_MSCCL=OFF"
```
Comment (gilbertlee-amd) [inline install.sh:215]: Shouldn't this be defining COMPILE_MSCCL?

**Review by gilbertlee-amd — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -17,6 +17,7 @@ option(BUILD_LOCAL_GPU_TARGET_ONLY             "Build only for GPUs detected on
 option(BUILD_SHARED_LIBS                       "Build as shared library"                       ON)
 option(BUILD_TESTS                             "Build unit test programs"                      OFF)
 option(COLLTRACE                               "Collective Trace Option"                       ON)
+option(ENABLE_MSCCL                            "Enable MSCCL while compiling"                  ON)
```
Comment (gilbertlee-amd) [inline CMakeLists.txt:20]: Rename this to ENABLE_MSCCL_KERNEL

File: CMakeLists.txt [hunk 3]
```diff
@@ -443,6 +431,39 @@ else()
...
+    src/include/msccl/msccl_status.h
+    src/include/msccl/msccl_struct.h
+    src/misc/msccl/msccl_lifecycle.cc
+    src/misc/msccl/msccl_parser.cc
+    src/misc/msccl/msccl_setup.cc
+    src/misc/msccl/msccl_status.cc
+  )
+endif()
+list(APPEND SRC_FILES ${MSCCL_SOURCES})
+
```
Comment (gilbertlee-amd) [inline CMakeLists.txt:466]: Move the common ones back into list of all sources, and only differentiate the kernel file.

File: CMakeLists.txt [hunk 4]
```diff
@@ -443,6 +431,39 @@ else()
 endif()
 list(APPEND SRC_FILES ${CU_SOURCES})
 
+if (ENABLE_MSCCL)
+  add_definitions(-DCOMPILE_MSCCL)
```
Comment (gilbertlee-amd) [inline CMakeLists.txt:435]: This line shouldn't be here.  It already is handled by target_compile_definitions(rccl PRIVATE COMPILE_MSCCL)

File: install.sh [hunk 5]
```diff
@@ -22,6 +22,7 @@ collective_trace=true
 enable_ninja=""
 install_dependencies=false
 install_library=false
+msccl_enabled=true
```
Comment (gilbertlee-amd) [inline install.sh:25]: compile_msccl_kernel

File: install.sh [hunk 6]
```diff
@@ -41,6 +42,7 @@ function display_help()
     echo "       --debug                 Build debug library"
     echo "       --disable_backtrace     Build without custom backtrace support"
     echo "       --disable-colltrace     Build without collective trace"
+    echo "       --disable-msccl         Build without MSCCL"
```
Comment (gilbertlee-amd) [inline install.sh:45]: disable_msccl_kernel

**Review by gilbertlee-amd — CHANGES_REQUESTED** [review]

File: install.sh [hunk 7]
```diff
@@ -208,6 +211,10 @@ if [[ "${collective_trace}" == false ]]; then
     cmake_common_options="${cmake_common_options} -DCOLLTRACE=OFF"
 fi
 
+if [[ "${msccl_kernel_enabled}" == false ]]; then
+    cmake_common_options="${cmake_common_options} -DENABLE_MSCCL_KERNEL=OFF"
+fi
+
```
Comment (gilbertlee-amd) [inline install.sh:217]: Why is this being defined?

**Review by gilbertlee-amd — CHANGES_REQUESTED** [review]

**Review by gilbertlee-amd — APPROVED** [review]

**Pull request merged**